README.md

```diff
@@ -0,0 +1,495 @@
+# sanctuary-def
+
+sanctuary-def facilitates the definition of curried JavaScript functions which
+are explicit about the number of arguments to which they may be applied and the
+types of those arguments.
+
+```javascript
+const $ = require('sanctuary-def');
+
+const def = $.create($.env);
```

Comment:
I've reworked the introduction. I believe the following paragraph captures the meaning conveyed by your suggested addition, Stefano:

> The next step is to define an environment. An environment is a list of [types](https://github.com/plaid/sanctuary-def/tree/dc-everything-2#types) such as `$.Number` and `$.String`. `$.env` is a default environment, which may be augmented with custom types:

```javascript
//    Integer :: Type
const Integer = ...;

//    NonZeroInteger :: Type
const NonZeroInteger = ...;

//    env :: [Type]
const env = $.env.concat([Integer, NonZeroInteger]);
```